cmake/dts.cmake

```
@@ -122,6 +122,10 @@ if(SUPPORTS_DTS)
   set(CACHED_DTS_ROOT_BINDINGS ${DTS_ROOT_BINDINGS} CACHE INTERNAL
     "DT bindings root directories")
 
+  if(NOT DEFINED CMAKE_DTS_PREPROCESS_C_COMPILER)
+    set(CMAKE_DTS_PREPROCESS_C_COMPILER ${CMAKE_C_COMPILER})
```

CMake uses the form: `CMAKE_<LANG>_COMPILER`
https://cmake.org/cmake/help/v3.0/variable/CMAKE_LANG_COMPILER.html

So let's just call it.
```suggestion
  if(NOT DEFINED CMAKE_DTS_COMPILER)
    set(CMAKE_DTS_COMPILER ${CMAKE_C_COMPILER})
```